apps/dot_ext/management/commands/authflow_id_delete_and_vacuume.py

```diff
@@ -0,0 +1,67 @@
+import time
+from django.db import connection
+
+from datetime import date, timedelta
+from django.core.management.base import BaseCommand
+
+from apps.dot_ext.models import AuthFlowUuidCopy
+
+DELETE_WITH_LIMIT = "DELETE FROM {table_name} WHERE auth_uuid IN (SELECT auth_uuid FROM {table_name} WHERE created < '{age_date}'::date ORDER BY created LIMIT {limit_on_delete})"
+VACUUM_TABLE = "VACUUM FULL {table_name}"
+
+
+def delete_and_vacuume_auth_uuid_table(age, limit_on_delete, run_vacuum, dry_run):
+    '''
+    Delete aged authflow uuid records, and then
+    call VACUUM FULL (Postgres only) on table: dot_ext_authflowuuid
+    '''
+
+    days_before = date.today()-timedelta(days=age)
+    # change table name dot_ext_authflowuuidcopy to dot_ext_authflowuuid before merge
+    delete_sql = DELETE_WITH_LIMIT.format(table_name="dot_ext_authflowuuidcopy", age_date=days_before.strftime("%m/%d/%Y"), limit_on_delete=limit_on_delete)
+    vacuum_sql = VACUUM_TABLE.format(table_name="dot_ext_authflowuuidcopy")
+
+    start_t = time.time()
+
+    aged_records_total = AuthFlowUuidCopy.objects.filter(created__lt=days_before).count()
+
+    print("Aged records total = {}".format(aged_records_total))
+
+    with connection.cursor() as cursor:
+
+        print("EXECUTE SQL : {}".format(delete_sql))
+
+        if not dry_run:
+            cursor.execute(delete_sql)
+
+        end_t = time.time()
+
+        print("Delete aged records: time elapsed (sec) = {}".format(end_t - start_t))
+
+        if run_vacuum:
+
+            print("EXECUTE SQL : {}".format(vacuum_sql))
+
+            if not dry_run:
+                cursor.execute(vacuum_sql)
+
+            end_t = time.time()
+            print("query + delete aged records and vacuum table: time elapsed (sec) = {}".format(end_t - start_t))
+
+
+class Command(BaseCommand):
+    help = ('Delete aged dot_ext_authflowuuid table records '
+            'and optionally vacuum the table - removing records marked as deleted (dead records) and reclaim space.')
+    
+    def add_arguments(self, parser):
+        parser.add_argument("-a", "--age", type=int, default=30, help="Age (in days) of authflow uuid records qualified for deletion.")
+        parser.add_argument("-l", "--limit", type=int, default=30000, help="Limit on number of delete allowed for one run.")

```

Comment:
Just curious - where did you come up with the 30k number?

Reply:
come up with 30k for below considerations:
1. Light enough to complete quick (no impact on normal traffic): the delete and vacuum will complete in very short time - in local sample run on 1000,000 simulated insert+delete (50%), and insert+stay (50%), it took < 2 sec
2. Large enough to drain backlog (500,000 for now), if the job is executed daily, a 30K quota will drain the backlog in less than 2 weeks time.